frontend/src/components/BatterySlider.jsx

```diff
@@ -0,0 +1,142 @@
+    const token = sessionStorage.getItem("accessToken")
+    fetch(API, {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${token}`,
+      },
+
+    })
+      .then((res) => {
+        if (!res.ok) {
+          throw new Error(res.statusText);
+        }
+        return res.json();
+      })
+      .then((data) => {
+        setFillPercentage(data.energyLevel)
+      })
+      .catch((error) => {
+        setError('There was a problem with the fetch operation: ' + error.message);
+      });
+  }, [id])
+
+  const handleDrag = (percentage) => {
+    const parsedPercent = parseInt(percentage, 10);
+    if (parsedPercent >= 0 && parsedPercent <= 100) {
+      setFillPercentage(parsedPercent)
+    }
+  }
+
+  const handleNewEnergy = async () => {
+    const token = sessionStorage.getItem("accessToken")
+    try {
+      const response = await fetch(API, {
+        method: "PATCH",
+        body: JSON.stringify({ energyLevel: fillPercentage }),
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: `Bearer ${token}`,
+        },
+      })
+      if (!response.ok) {
+        throw new Error("Failed to update data")
+      }
+      const updatedData = await response.json()
+      if (
+        updatedData.energyLevel !== undefined &&
+        updatedData.energyLevel !== null
+      ) {
+        setFillPercentage(updatedData.energyLevel)
+        console.log(updatedData)
+      }
+    } catch (error) {
+      console.error("Error updating energy data:", error)
+    }
+  }
+
+
+  // then we update energy level with a patch request
+  const updateEnergyLevel = (value) => {
+    //convert the input value string to number
+    const parsedValue = parseInt(value, 10);
+    const token = sessionStorage.getItem("accessToken")
+    fetch(API, {
+      method: "PATCH",
+      body: JSON.stringify({ energyLevel: parsedValue }),
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${token}`
+      },
+    })
+      .then((res) => {
+        if (!res.ok) {
+          throw new Error(res.statusText);
+        }
+        return res.json();
+      })
+      .then((data) => {
+        //checking what we get back from the backend
+        console.log(data)
+        // updating the state with the new value
+        if (data.success) {
+          setFillPercentage(value)
+          setMessage(`Energy level updated with ${value}%`)
+        }
+      })
+      .catch((error) => {
+        setError('Error: ' + error.message);
+      });
+  }
+
+  return (
+    <>
+
+      <BatterySliderWrapper id={id}>
+        <BatterySVG fillPercentage={fillPercentage} onDrag={handleDrag} />
+        <input
+          type="range"
+          min="0"
+          max="100"
+          value={fillPercentage}
+          onChange={(e) => updateEnergyLevel(e.target.value)}
+        />
+        <div>Current energy level: {fillPercentage}%</div>
```

Comment:
Yes I think it's smart to say the percentage, Let's see how to implement it in the design